### PR TITLE
Integration 4/25 - wjunhao/gitflow_230425_o3de_extras

### PR DESCRIPTION
**Pull request by forkercat**
Signed-off-by: Junhao Wang <wjunhao@amazon.com>

## What does this PR do?

This PR integrates changes from stabilization/2305 (https://github.com/o3de/o3de-extras/commit/894947f3b75914e6ed4483feb8f8b09628d0a781) into development as of 4/25/2023.

**Run the merge command:**

```bash
git log --reverse --format=%H --first-parent upstream/development..upstream/stabilization/2305 | ./xargs git merge
```

**There are 2 merge conflicts in `ROS2ModuleInterface.h` file:**

They are all related to 'include' files and descriptor list configuration.

- https://github.com/o3de/o3de-extras/commit/eac24a42eb0a28ee2d7600009d9947e8ac38f727 (added `ROS2Lidar2DSensorComponent` include line and descriptor)
- https://github.com/o3de/o3de-extras/commit/df57dcce6484b5cf536a4f58524eb1865809772c (picked dev - might needs some eyes on this)

### CODE REVIEW

File: Gems/ROS2/Code/Source/ROS2ModuleInterface.h [hunk 1]
```diff
@@ -58,6 +60,7 @@ namespace ROS2
                     ROS2ImuSensorComponent::CreateDescriptor(),
                     ROS2GNSSSensorComponent::CreateDescriptor(),
                     ROS2LidarSensorComponent::CreateDescriptor(),
+                    ROS2Lidar2DSensorComponent::CreateDescriptor(),
```
Comment (michalpelka) [inline Gems/ROS2/Code/Source/ROS2ModuleInterface.h:63]: ROS2WheelOdometry seems to be missing.

Comment (forkercat) [reply Gems/ROS2/Code/Source/ROS2ModuleInterface.h:63]: Do you know if these modules are still needed?

```cpp
JointPublisherComponent::CreateDescriptor(),
ManipulatorControllerComponent::CreateDescriptor(),
PidMotorControllerComponent::CreateDescriptor(),
MotorizedJointComponent,
```

Comment (michalpelka) [reply Gems/ROS2/Code/Source/ROS2ModuleInterface.h:63]: This is part of features that are in development only and those are needed.
MotorizedJointComponent was refactored and currently lives only in stabilization.

Comment (forkercat) [reply Gems/ROS2/Code/Source/ROS2ModuleInterface.h:63]: Thanks for the info. I'll include those in dev and remove `MotorizedJointComponent`.

**Review by adamdbrw — APPROVED** [review]

**Comment by forkercat** [comment]
I notice there are DCO issues. I can skip the check for this PR, but it would be good to have the commits signed off correctly in the future.

> There are 9 commits incorrectly signed off. This means that the author(s) of these commits failed to include a Signed-off-by line in their commit message.
> To avoid having PRs blocked in the future, always include Signed-off-by: Author Name <authoremail@example.com> in every commit message. You can also do this automatically by using the -s flag (i.e., git commit -s).

image

fyi, Looks like the emails do not match. @arturkamieniecki @MateuszWasilewski @pawelbudziszewski @ZuzaOsa

**Comment by michalpelka** [comment]
I've tested it, and it fails to compile : https://gist.github.com/michalpelka/608e812df0e6385b04fdb7d69a039682

**Comment by michalpelka** [comment]
It is not compiling for me:
https://gist.github.com/michalpelka/34ab1197a2f36a518592070cb1480a7c
I will later update you what was issue.

**Comment by zakmat** [comment]
I have created PR https://github.com/aws-lumberyard-dev/o3de-extras/pull/1 to address the compilation issues

**Review by gadams3 — APPROVED** [review]

**Pull request merged**